support uri in getinfo response

wilmerpaulino opened the pull request:
Fixes #550.

I generated the new `protobuf` definitions as stated in the [code contribution guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#Protobuf).

I'm fairly new to the project, so I'm not sure what's the best way to determine the node's public IP address. I'm submitting this initial implementation for some feedback.

cfromknecht commented on lnrpc/rpc.swagger.json:
```diff
@@ -696,6 +696,15 @@
     }
   },
   "definitions": {
+    "NewAddressRequestAddressType": {
```
Can't see why this would have changed, perhaps rebasing will fix this?

halseth commented on rpcserver.go:
```diff
@@ -1113,6 +1113,18 @@ func (r *rpcServer) GetInfo(ctx context.Context,
+	var hostPort net.Addr
+	addrs := r.server.currentNodeAnn.Addresses
+	if len(addrs) != 0 {
+		hostPort = addrs[0]
```
Since it can have more that one external IP, we can just return a list of all.

wilmerpaulino replied:
This means we would end up having a list of URIs as well.

halseth replied:
That's probably useful :) Take a look at `lncli getnodeinfo`

wilmerpaulino commented:
Running `gen_protos.sh` on the current master branch results in the following [diff](https://gist.github.com/wilmerpaulino/3a0e447826379683378764a3f115063d). These seem to be the changes that were applied to my commit. Should a new PR be opened to address this?

halseth commented:
What version of `protoc` are you using?

wilmerpaulino commented:
```bash
$ protoc --version
libprotoc 3.4.0
```

halseth commented:
Try updating, and see if that helps. I'm using `3.5.0`

wilmerpaulino commented:
Installed version `3.5.1` from my package manager and it results in the same diff. I was using `3.4.0` since that's what's stated in the [code contribution guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#Protobuf).

wilmerpaulino commented:
Sorry about closing the PR, had some issues with git. I figured out what I was missing when generating the new `protobuf` definitions. I'll be opening a separate PR to clarify some parts in the documentation.

Rebased to master and addressed comments.

halseth commented on rpcserver.go:
```diff
@@ -1113,9 +1114,17 @@ func (r *rpcServer) GetInfo(ctx context.Context,
 
+	// Check if external IP addresses were provided to lnd and use them
+	// to set the URIs.
+	addrs := r.server.currentNodeAnn.Addresses
```
I don't think this is thread safe, since the current node announcement can be changed by another thread in the server. Instead you could do 
```go
nodeAnn, err := r.server.genNodeAnnouncement(false)
...
```
do safely fetch a copy of the node announcement :)

cfromknecht commented:
Thanks for fixing @wilmerpaulino! I too am running 3.4.0, compiled protos myself, and tested locally. LGTM ⚡️
